packages/workflow/src/workflow.ts

```diff
@@ -1253,3 +1253,20 @@ export function upsertSearchAttributes(searchAttributes: SearchAttributes): void
 
 export const stackTraceQuery = defineQuery<string>('__stack_trace');
 export const enhancedStackTraceQuery = defineQuery<EnhancedStackTrace>('__enhanced_stack_trace');
+
+const loggerSinks = proxySinks<LoggerSinks>();
+
+/**
+ * Default workflow logger.
+ * This logger is replay-aware and will omit log messages on workflow replay.
+ * The messages emitted by this logger are funnelled to the worker's `defaultSinks`, which are installed by default.
+ *
+ * Note that since sinks are used to power this logger, any log attributes must be transferable via the
+ * {@link https://nodejs.org/api/worker_threads.html#worker_threads_port_postmessage_value_transferlist | postMessage}
+ * API.
+ *
+ * `defaultSinks` accepts a user logger and defaults to the `Runtime`'s logger.
+ *
+ * See the documentation for `WorkerOptions`, `defaultSinks`, and `Runtime` for more information.
+ */
+export const log = loggerSinks.defaultWorkerLogger;
```

Comment:
I'm thinking that maybe we should clone arguments (ie. `JSON.parse(JSON.stringify(args))`) on Workflow side (before they get captured by the sink plumbing). I got bitten once because an object that was being passed to a sink was being modified afterward (but in the same WFT), so the sink would receive the object in its modified state rather than the object in the state it was at the moment the sink was called in code...

Reply:
For similar reason, wondering if we should maybe capture the _real_ time at the moment `log` was called...

Reply:
For now, I suggest we simply mark this as experimental. Let's get this in as is, and we'll see later if either of my previous concern turns out to be justified.

Reply:
I wouldn't want to impose JSON serializability here, `postMessage` accepts a wider variety of types.

Reply:
Why not just use this which is mentioned in the linked docs? https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm

I think the point James makes seems pretty important

Reply:
I agree that the point is important, I’ll look into it

Reply:
All good points @mjameswh , thanks